server/data/redisClient.ts

```diff
@@ -9,10 +10,22 @@ const url =
     ? `rediss://${config.redis.host}:${config.redis.port}`
     : `redis://${config.redis.host}:${config.redis.port}`
 
-export const createRedisClient = (legacyMode = false): RedisClient => {
-  return createClient({
+export const createRedisClient = ({ legacyMode }: { legacyMode: boolean }): RedisClient => {
+  const client = createClient({
     url,
     password: config.redis.password,
     legacyMode,
+    socket: {
+      reconnectStrategy: (attempts: number) => {
+        // Exponential back off: 20ms, 40ms, 80ms..., capped to retry every 30 seconds
+        const nextDelay = Math.min(2 ** attempts * 20, 30000)
+        logger.info(`Retry Redis connection attempt: ${attempts}, next attempt in: ${nextDelay}ms`)
+        return nextDelay
+      },
+    },
```

Comment:
The default reconnect strategy in v4 continually retries to connect which maxes out CPU [bug](https://github.com/redis/node-redis/issues/2095)